pkg/engine/lifecycletest/fuzz_test.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lifecycletest
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"testing"
+
+	"github.com/pulumi/pulumi/pkg/v3/engine/lifecycletest/fuzzing"
+	"github.com/pulumi/pulumi/pkg/v3/resource/stack"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/apitype"
+	"github.com/stretchr/testify/require"
+	"pgregory.net/rapid"
+)
+
+// TestFuzz is a harness for running fuzz tests against the lifecycle test framework. It will generate random lifecycle
+// test fixtures and run them, hunting for possible snapshot integrity errors.
+//
+// Usage:
+//
+// This test will be skipped unless the `PULUMI_LIFECYCLE_TEST_FUZZ` environment variable is set. This is because
+// fuzzing is by its nature potentially flaky, and so we want fine-grained control of when we run fuzzing, especially in
+// CI where a red build can prevent a PR from being merged.
+//
+// If you want to customize the fuzzing that occurs, you can modify (but not commit!) the FixtureOptions passed to
+// fuzzing.GeneratedFixture in this test to your needs.
+func TestFuzz(t *testing.T) {
+	t.Parallel()
+
+	shouldFuzz := os.Getenv("PULUMI_LIFECYCLE_TEST_FUZZ")
```

Comment:
Instead of doing this, would it be worth moving the fuzz tests to a separate package which isn't run in CI by default?  Might make it a little more convenient to run directly locally.  Not a big deal though, so feel free to ignore.